dotnet/sdk-phase-2.md

```diff
@@ -0,0 +1,484 @@
+
+    public static Task<ChildWorkflowHandle<TResult>> StartChildWorkflowAsync<T, TResult>(
+        Func<T, Task<TResult>> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle> StartChildWorkflowAsync(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle<TResult>> StartChildWorkflowAsync<TResult>(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync(
+        Func<Task> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync<T>(
+        Func<T, Task> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<TResult>(
+        Func<Task<TResult>> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<T, TResult>(
+        Func<T, Task<TResult>> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<TResult>(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    // Signal and query handlers
+
+    public bool TryGetSignalHandler(string name, out Delegate handler) { /*...*/ }
+
+    public void SetSignalHandler(string name, Func<Task> handler) { /*...*/ }
+
+    public void SetSignalHandler<T>(string name, Func<T, Task> handler) { /*...*/ }
+
+    public void RemoveSignalHandler(string name) { /*...*/ }
+
+    public bool TryGetDynamicSignalHandler(out Func<string, Payload[], Task> handler) { /*...*/ }
+
+    public void SetDynamicSignalHandler(Func<string, Payload[], Task> handler) { /*...*/ }
+
+    public void RemoveDynamicSignalHandler() { /*...*/ }
+
+    public bool TryGetQueryHandler(string name, out Delegate handler) { /*...*/ }
+
+    public void SetQueryHandler<TResult>(string name, Func<TResult> handler) { /*...*/ }
+
+    public void SetQueryHandler<T, TResult>(string name, Func<T, TResult> handler) { /*...*/ }
+
+    public void RemoveQueryHandler(string name) { /*...*/ }
+
+    public bool TryGetDynamicQueryHandler<TResult>(out Func<string, Payload[], TResult> handler) { /*...*/ }
+
+    public void SetDynamicQueryHandler<TResult>(Func<string, Payload[], TResult> handler) { /*...*/ }
+
+    public void RemoveDynamicSignalHandler() { /*...*/ }
+
+    // Search attributes and memo
+
+    public static SearchAttributes SearchAttributes { get; }
+
+    public static void UpdateSearchAttributes(params SearchAttributeUpdate[] updates) { /*...*/ }
+
+    public static IDictionary<string, Payload> RawMemo { get; }
```

Comment:
Can we have a memo getter that doesn't return raw payloads, something that will embed the configured data converter so users don't have to decode payloads themselves?

Reply:
Yes, from the notes after this code:

```
* We may add helpers for getting individual memo values given a type to deserialize to
  * We don't expose a data converter to the user in a workflow at this time, so it is wise to make these helpers
```

Reply:
Added explicit API for this